examples/kyb-app/src/components/organisms/UIRenderer/elements/JSONForm/components/DocumentField/DocumentField.tsx

```diff
@@ -0,0 +1,117 @@
+import { useStateManagerContext } from '@app/components/organisms/DynamicUI/StateManager/components/StateProvider';
+import { useUIElementToolsLogic } from '@app/components/organisms/DynamicUI/hooks/useUIStateLogic/hooks/useUIElementsStateLogic/hooks/useUIElementToolsLogic';
+import { Document, UIElement } from '@app/domains/collection-flow';
+import { fetchFile, uploadFile } from '@app/domains/storage/storage.api';
+import { AnyObject, ErrorsList, FileInputAdapter, RJSVInputProps } from '@ballerine/ui';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import set from 'lodash/set';
+import { collectionFlowFileStorage } from '@app/pages/CollectionFlow/collection-flow.file-storage';
+import { useUIElementState } from '@app/components/organisms/UIRenderer/hooks/useUIElementState';
+import { useUIElementErrors } from '@app/components/organisms/UIRenderer/hooks/useUIElementErrors/useUIElementErrors';
+
+interface DocumentFieldParams {
+  documentData: AnyObject & { id: string };
+  mappingParams: {
+    documentIndex: number;
+    documentPage: number;
+  };
+}
+
+export const DocumentField = (
+  props: RJSVInputProps & { definition: UIElement<DocumentFieldParams> },
+) => {
+  const { definition, ...restProps } = props;
+  const { onChange } = restProps;
+  const { stateApi } = useStateManagerContext();
+  const { payload } = useStateManagerContext();
+  const { options } = definition;
+  const [fileItem, setFile] = useState<File | null>(null);
+
+  const { toggleElementLoading } = useUIElementToolsLogic(definition.name);
+  const { state: elementState } = useUIElementState(definition);
+  const documentDefinition = useMemo(
+    () => ({
+      ...definition,
+      valueDestination: `document-error-${definition.options.documentData.id}`,
+    }),
+    [definition],
+  );
```

Comment:
Redundant `useMemo`.

Reply:
Its not reduntant.As i mentioned in `useMemo` response before.Non memoized creation of object will cause re-calculation in every dependant hook.So calculations will be run on each re-render and we dont want that.